src/Services/FHIR/FhirServiceRequestService.php

```diff
@@ -0,0 +1,410 @@
+            $codeableConcept->addCoding(
+                UtilsService::createCoding(
+                    $dataRecord['standard_code'],
+                    $description,
+                    FhirCodeSystemConstants::LOINC
+                )
+            );
+        }
+
+        // If no codes, use text only or data absent
+        if (empty($codeableConcept->getCoding())) {
+            if (!empty($dataRecord['procedure_name'])) {
+                $codeableConcept->setText($dataRecord['procedure_name']);
+            } else {
+                return UtilsService::createDataAbsentUnknownCodeableConcept();
+            }
+        }
+
+        return $codeableConcept;
+    }
+
+    /**
+     * Build reason codes from order_diagnosis field
+     * Format: "code1:type1;code2:type2"
+     */
+    private function buildReasonCodes($diagnosisString)
+    {
+        $reasonCodes = [];
+        $codesService = new CodeTypesService();
+
+        $diagnoses = explode(";", $diagnosisString);
+        foreach ($diagnoses as $diagnosis) {
+            if (empty(trim($diagnosis))) {
+                continue;
+            }
+
+            $parts = explode(":", $diagnosis);
+            if (count($parts) >= 2) {
+                $code = trim($parts[0]);
+                $codeType = trim($parts[1]);
+
+                $fullCode = $codesService->getCodeWithType($code, $codeType);
+                $description = $codesService->lookup_code_description($fullCode);
+                $system = $codesService->getSystemForCodeType($codeType);
+
+                if (!empty($system)) {
+                    $reasonCodes[] = UtilsService::createCodeableConcept([
+                        $code => [
+                            'code' => $code,
+                            'description' => $description ?: null,
+                            'system' => $system
+                        ]
+                    ]);
+                }
+            }
+        }
+
+        return $reasonCodes;
+    }
+
+    /**
+     * Creates the Provenance resource for the ServiceRequest
+     */
+    public function createProvenanceResource($dataRecord, $encode = false)
+    {
+        if (!($dataRecord instanceof FHIRServiceRequest)) {
+            throw new \BadMethodCallException("Data record should be correct instance class");
+        }
+
+        $fhirProvenanceService = new FhirProvenanceService();
+        $requester = $dataRecord->getRequester();
+
+        $fhirProvenance = $fhirProvenanceService->createProvenanceForDomainResource(
+            $dataRecord,
+            $requester
+        );
+
+        if ($encode) {
+            return json_encode($fhirProvenance);
+        }
+
+        return $fhirProvenance;
+    }
+
+    /**
+     * Returns the Canonical URIs for US Core Implementation Guide Profiles
+     */
+    public function getProfileURIs(): array
+    {
+        return [
+            'http://hl7.org/fhir/us/core/StructureDefinition/us-core-servicerequest'
```

Comment:
I can add in the work for this as its in my other PR, but we need to add versioned URL's to the profiles.  So we'd have 3 profiles here.  Unversioned like you have here and then 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-servicerequest|8.0.0', 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-servicerequest|7.0.0'.